### PR TITLE
feat: Allow multi party reconciliation against a Bank Transaction

### PR DESCRIPTION
**Pull request by marination**
- Use a Journal Entry to create a party agnostic settlement
- ui: Popup to warn user and ask for JE account in case of multi party selection. The **Debit To**, **Credit To** or **Payable Account** is fetched by the backend depending on the kind of vouchers selected
- 2024-10-10 12 31 35 PM
-  Screenshot 2024-10-09 at 6 47 59 PM
-  UI Validation (as well as backend, that already existed) against selecting vouchers of different types
-  Screenshot 2024-10-10 at 12 38 19 PM

**ToDo:**

- [x] Code cleanup and abstractions
- [x] Adjust tests

### CODE REVIEW

**Comment by marination** [comment]
@barredterra Do we need a setting here since the popup already gives the user a heads up. If someone does not desire this and it was a mistake they should just shut the dialog. Or do we want to make sure this not accessible if a company does not need it ?

**Comment by barredterra** [comment]
@marination I didn't look at the code yet, but from the screenshots I wondered: shouldn't we use the voucher's _Debit To_ / _Credit To_ accounts, instead of letting the user set one manually?

File: banking/overrides/bank_transaction.py [hunk 1]
```diff
@@ -282,6 +385,20 @@ def get_outstanding_amount(payment_doctype, payment_name) -> float:
...
 
+def get_debtor_creditor_account(invoice: dict) -> str | None:
+	"""Get the debtor or creditor (intermediate) account based on the invoice type."""
+	account_field = (
+		"debit_to"
+		if invoice.get("voucher_type") == "Sales Invoice"
+		else "credit_to"
+		if invoice.get("voucher_type") == "Purchase Invoice"
+		else "payable_account"
+	)
```
Comment (barredterra) [inline banking/overrides/bank_transaction.py]:
```suggestion
		if invoice.get("voucher_type") == "Sales Invoice":
			account_field = "debit_to"
		elif invoice.get("voucher_type") == "Purchase Invoice":
			account_field = "credit_to"
		else:
			account_field = "payable_account"
```

File: banking/public/js/bank_reconciliation_beta/actions_panel/match_tab.js [hunk 2]
```diff
@@ -253,6 +275,25 @@ erpnext.accounts.bank_reconciliation.MatchTab = class MatchTab {
...
+			},
+		]
+		frappe.prompt(
+			fields,
+			() => {
+				me.bulk_reconcile_vouchers(selected_vouchers, true);
+			},
+			__("Multiple Party Reconciliation"),
+			__("Confirm")
+		);
```
Comment (barredterra) [inline banking/public/js/bank_reconciliation_beta/actions_panel/match_tab.js]:
```suggestion
		frappe.confirm(
			__("Are you trying to reconcile vouchers of different parties? This action will reconcile vouchers using a Journal Entry."),
			() => this.bulk_reconcile_vouchers(selected_vouchers, true);
		);
```

**Comment by barredterra** [comment]
@Mergifyio backport version-14

**Pull request merged**